### PR TITLE
KAFKA-3473; More Controller Health Metrics (KIP-237)

### PR DESCRIPTION
**Pull request by lindong28**
This patch adds a few metrics that are useful for monitoring controller health. See KIP-237 for more detail.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by lindong28** [comment]
@ijuma Not sure if you have time to review this patch. Do you have any recommendation?

**Comment by lindong28** [comment]
@becketqin Just in case you have time, can you review this patch?

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-test-coverage/299/

**Comment by lindong28** [comment]
@junrao @ijuma I will update the patch if we want to include these new sensors in 1.1 release. Otherwise never mind.

File: core/src/main/scala/kafka/controller/KafkaController.scala [hunk 1]
```diff
@@ -1554,6 +1554,8 @@ private[controller] class ControllerStats extends KafkaMetricsGroup {
 }
 
 sealed trait ControllerEvent {
+  @volatile var enqueueTimeMs: Long = -1L
```
Comment (junrao) [inline core/src/main/scala/kafka/controller/KafkaController.scala]: Does enqueueTimeMs need to be var? It seems that we can make it a val and initialize it with Time.SYSTEM.milliseconds().

**Review by junrao — COMMENTED** [review]
@lindong28 : Thanks for the patch. LGTM. Just a minor comment below.

File: core/src/main/scala/kafka/controller/KafkaController.scala (hunk 1 above)
Comment (lindong28) [reply core/src/main/scala/kafka/controller/KafkaController.scala]: Good point. Thanks for the review! I have updated the patch as suggested.

**Review by junrao — APPROVED** [review]
@lindong28 : Thanks for the updated patch. LGTM. You can merge it after the tests pass.

**Comment by lindong28** [comment]
@junrao It is merged now. Thanks!

**Comment by ijuma** [comment]
kafka.metrics.MetricsTest.testMetricsLeak is failing frequently. Could it be related to this? One example:

https://builds.apache.org/job/kafka-trunk-jdk8/2655/testReport/kafka.metrics/MetricsTest/testMetricsLeak/

**Comment by lindong28** [comment]
@ijuma This patch metrics in controller and the number of new metrics does not depend on client request. Since testMetricsLeak does not bounce broker, the test failure is probably not due to this patch. I opened https://github.com/apache/kafka/pull/5062 to print the metric name to help debug this flaky test.

**Pull request closed without merging**